projects/ngneat/transloco-locale/src/lib/shared.ts

```diff
@@ -0,0 +1,8 @@
+import { Locale } from './transloco-locale.types';
+import { LocaleSettings, LocaleConfig } from './transloco-locale.config';
+
+export function getDefaultOptions(locale: Locale, type: keyof LocaleSettings, localeConfig: LocaleConfig): any {
+  const defaultConfig = localeConfig.global[type] || {};
+  const settings = localeConfig.localeBased[locale];

```

Comment:
You should merge the given `localeConfig` with the `global/localeBased` as someone can override it and pass a partial config in the component's providers or a lazy load module.